remoulade/state/backend.py

```diff
@@ -0,0 +1,62 @@
+from typing import Iterable
+
+from ..encoder import Encoder
+
+
+class State:
+    """Catalog Class, that
+    storages the state payload
+    """
+    def __init__(self, state):
+        self.state = state
+
+    def _asdict(self):
+        return {"state": self.state}
+
+
+Missing = type("Missing", (object,), {})()
+
+
+class StateBackend:
+    """ABC for  state backends.
+
+    Parameters:
+      namespace(str): The logical namespace under which the data
+        should be stored.
+      encoder(Encoder): The encoder to use when storing and retrieving
+        result data.  Defaults to :class:`.JSONEncoder`.
+    """
+
+    def __init__(self, *, namespace: str = "remoulade-results", encoder: Encoder = None):
```

Comment:
ok basically the method I want to see defined in this class are : 
- `get_status`
- `store_status`
- `build_message_key`

Also namespace should be `remoulade-status`, the use of a namescpace is to be sure data stored by StatusBackend are not erased by ResultBackend.

You could add a superclass backend where you want to